apps/transport/lib/jobs/dataset_quality_score.ex

```diff
@@ -0,0 +1,281 @@
+defmodule Transport.Jobs.DatasetQualityScoreDispatcher do
+  @moduledoc """
+  Computes quality scores for a dataset
+  """
+  use Oban.Worker, unique: [period: 60 * 60 * 20], max_attempts: 1
+  import Ecto.Query
+
+  @impl Oban.Worker
+  def perform(_job) do
+    DB.Dataset.base_query()
+    |> DB.Repo.all()
+    |> Enum.map(fn dataset ->
+      %{dataset_id: dataset.id} |> Transport.Jobs.DatasetQualityScore.new()
+    end)
+    |> Oban.insert_all()
+
+    :ok
+  end
+end
+
+defmodule Transport.Jobs.DatasetQualityScore do
+  @moduledoc """
+  Computes quality scores for a dataset
+  """
+  use Oban.Worker, unique: [period: 60 * 60 * 20], max_attempts: 1
+  import Ecto.Query
+
+  @impl Oban.Worker
+  def perform(%Oban.Job{args: %{"dataset_id" => dataset_id}}) do
+    save_dataset_freshness_score(dataset_id)
+    :ok
+  end
+
+  def save_dataset_freshness_score(dataset_id) do
+    %{score: score, details: details} = dataset_freshness_score(dataset_id)
+
+    %DB.DatasetScore{
+      dataset_id: dataset_id,
+      topic: "freshness",
+      score: score,
+      timestamp: DateTime.utc_now(),
+      details: details
+    }
+    |> DB.Repo.insert()
+  end
+
+  @spec dataset_freshness_score(integer) :: %{details: map, score: nil | float}
+  @doc """
+  dataset "freshness" is the answer to the question:
+  "When the data was downloaded, was it up-to-date?"
+  To give a score, we proceed this way:
+  - get the dataset's current resources
+  - for each resource, give it a score
+  - we compute an average of those scores to get a score at the dataset level
+  - that score is averaged with the dataset's last computed score, using exponential smoothing
+  (see the function exp_smoothing below). This allows a score to reflect not only the current
+  dataset situation but also past situations. Typically, a dataset that had outdated resources
+  for the past year, but only up-to-date resources today is expected to have a low freshness score.
+  The interest of exponential smoothing is to give past scores an increasingly small weight as time
+  passes. To have a good score, a dataset must have up-to-date resources every day.
+  """
+  def dataset_freshness_score(dataset_id) do
+    %{dataset_freshness: today_score, details: details} = current_dataset_freshness(dataset_id)
+    last_dataset_freshness = last_dataset_freshness(dataset_id)
+
+    computed_score =
+      case last_dataset_freshness do
+        %{score: previous_score} when is_float(previous_score) ->
+          exp_smoothing(previous_score, today_score)
+
+        _ ->
```

Comment:
```suggestion
        nil ->
```
Ou possible un autre cas ?

Reply:
`%{score: nil}` ou `nil` sont possibles (un score en base, mais sans valeur ou pas de score en base)